tenzoMath/fanucModel.cpp

```diff
@@ -0,0 +1,326 @@
+                     * sin(param[2]._alphaParam);
+    const double e = param[1]._aParam * param[1]._aParam + param[2]._aParam * param[2]._aParam + param[1]._dParam *
+                     param[1]._dParam + param[2]._dParam * param[2]._dParam + param[3]._dParam * param[3]._dParam -
+                     param[0]._aParam * param[0]._aParam - coord[0] * coord[0] - coord[1] * coord[1] - 
+                     (coord[2] - param[0]._dParam) * (coord[2] - param[0]._dParam) + 2 *
+                     param[1]._dParam * param[2]._dParam * cos(param[1]._alphaParam) + 2 * param[1]._dParam * param[3]._dParam *
+                     cos(param[1]._alphaParam) * cos(param[2]._alphaParam) + 2 * param[2]._dParam * param[3]._dParam * cos(param[2]._alphaParam);
+    const double f = coord[1] * sin(param[0]._alphaParam);
+    const double g = -coord[0] * sin(param[0]._alphaParam);
+    const double h = -param[3]._dParam * sin(param[1]._alphaParam) * sin(param[2]._alphaParam);
+    const double i = param[2]._aParam * sin(param[1]._alphaParam);
+    const double j = param[1]._dParam + param[2]._dParam * cos(param[1]._alphaParam) + param[3]._dParam *
+                     cos(param[1]._alphaParam) * cos(param[2]._alphaParam) - (coord[2] - param[0]._dParam) *
+                     cos(param[0]._alphaParam);
+    const double r = 4. * param[0]._aParam * param[0]._aParam * (j - h) * (j - h) + sin(param[0]._alphaParam) *
+                     sin(param[0]._alphaParam) * (e - c) * (e - c)
+                     - 4. * param[0]._aParam * param[0]._aParam * sin(param[0]._alphaParam) * sin(param[0]._alphaParam)
+                     * (coord[0] * coord[0] + coord[1] * coord[1]);
+    const double s = 4. * (4. * param[0]._aParam * param[0]._aParam * i * (j - h) + sin(param[0]._alphaParam) *
+                           sin(param[0]._alphaParam) * d *
+                           (e - c));
+    const double t = 2. * (4. * param[0]._aParam * param[0]._aParam * (j * j - h * h + 2 * i * i) +
+                           sin(param[0]._alphaParam) * sin(param[0]._alphaParam)
+                           * (e * e - c * c + 2 * d * d) - 4. * param[0]._aParam * param[0]._aParam *
+                           sin(param[0]._alphaParam) * sin(param[0]._alphaParam) *
+                           (coord[0] * coord[0] + coord[1] * coord[1]));
+    const double u = 4. * (4. * param[0]._aParam * param[0]._aParam * i * (j + h) +
+                           sin(param[0]._alphaParam) * sin(param[0]._alphaParam) * d * (e + c));
+    const double v = 4. * param[0]._aParam * param[0]._aParam * (h + j) * (h + j) + sin(param[0]._alphaParam) *
+                     sin(param[0]._alphaParam) *
+                     (e + c) * (e + c) - 4. * param[0]._aParam * param[0]._aParam * sin(param[0]._alphaParam) *
+                     sin(param[0]._alphaParam) *
+                     (coord[0] * coord[0] + coord[1] * coord[1]);
```

Comment:
![MemeFormulae](https://memepedia.ru/wp-content/uploads/2017/05/%D1%87%D1%83%D0%B2%D0%B0%D0%BA-%D0%B8%D0%B7-%D0%BC%D0%B0%D0%BB%D1%8C%D1%87%D0%B8%D1%88%D0%BD%D0%B8%D0%BA%D0%B0-%D0%BC%D0%B5%D0%BC.jpg)

(**must be changed**): Mmm.. unreadable. You exactly need to refactor THIS code.